core/scripts/chaincli/handler/handler.go

```diff
@@ -202,15 +202,17 @@ func (h *baseHandler) waitDeployment(ctx context.Context, tx *ethtypes.Transacti
 	}
 }
 
-func (h *baseHandler) waitTx(ctx context.Context, tx *ethtypes.Transaction) {
+func (h *baseHandler) waitTx(ctx context.Context, tx *ethtypes.Transaction) bool {
```

Comment:
Maybe we could change this signature to return an error instead of a bool? So:

```golang
func (h *baseHandler) waitTx(ctx context.Context, tx *ethtypes.Transaction) error
```

And then, inside the function body, when err != nil, return err, and when `receipt.Status == ethtypes.ReceiptStatusFailed`, return errors.New("transaction failed")?